dev-apps/js.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3.6
+# -*- coding: utf-8 -*-
+
+import os
+import stdlib
+from stdlib.template import autotools
+from stdlib.template.configure import configure
+from stdlib.manifest import manifest
+
+
+@manifest(
+    name='js',
+    category='dev-apps',
+    description='''
+    JavaScript interpreter and libraries - Version 60.
+    ''',
+    tags=['JavaScript', 'Mozilla', 'JS'],
+    maintainer='dorian.trubelle@epitech.eu',
+    licenses=[stdlib.license.License.MOZILLA],
+    upstream_url='https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey',
+    kind=stdlib.kind.Kind.EFFECTIVE,
+    versions_data=[
+        {
+            'semver': '60.8.0',
+            'fetch': [{
+                'url': 'http://ftp.gnome.org/pub/gnome/teams/releng/tarballs-needing-help/mozjs/mozjs-60.8.0.tar.bz2',
+                'sha256': '697331336c3d65b80ded9ca87b4a8ceb804e5342b476eaa133ac638102a9dc5d',
+            },
+            ],
+        },
+    ],
+    build_dependencies=[
+        'dev-libs/icu-dev',
+        'sys-apps/which',
+        'dev-lang/python2',
+        'sys-libs/readline-dev',
+        'dev-apps/autoconf2-13'
+    ]
+)
+def build(build):
+    packages = autotools.build(
+        # export SHELL is required because we're building the package in the chroot environment
```

Comment:
Try to replace all that shell code into python code, to make things a bit smoother to read.

You can use `os.environ['SHELL'] = '/bin/bash'` at the beginning of the manifest, for example. Or even the argument `build_folder` of `autotools.build()` to automatically create the build folder. Doing so will remove the need of the other `stdlib.cmd` used in the compile and install steps, as it already goes into that directory before executing those steps (see the documentation of [`autotools.build()`](https://docs.raven-os.org/p/nbuild/master/stdlib.template.autotools.html) )

After that, you can try to use the default `configure` template and set the `binary` argument to `../js/src/configure` and see if it works. 

If it does, remove the `--prefix=/usr` argument, as it will be automatically provided by the wrapper template.

If it doesn't, please include the return value of [`stdlib.template.configure.get_dir_flags`](https://docs.raven-os.org/p/nbuild/master/stdlib.template.configure.html#stdlib.template.configure.get_dir_flags) as a remplacement of the `--prefix=/usr` argument.

Reply:
setting `binary` didn't work, so I've tried using the return value of `stdlib.template.configure.get_dir_flags`. The problem is that it also returned the flag ` --exec-prefix`, which isn't recognized by the configure script of this package, thus raising an error. So I've left the flag `--prefix=/usr`